API/Controllers/AdminNotesController.cs

```diff
@@ -0,0 +1,168 @@
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+using API.Authorization;
+using API.DTOs;
+using API.Services.Implementations;
+using API.Services.Interfaces;
+using API.Utilities.AdminNotes;
+using API.Utilities.Extensions;
+using Asp.Versioning;
+using Database;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+/*
+ * This controller directly uses the context to check for existence of entities
+ *
+ * (AFAIK - myssto) Without extensive extra infrastructure or breaking the completely dynamic
+ * design pattern of this tree, it would not be possible to access the services or repositories
+ * of the parent entity types to satisfy our existing convention of never accessing the context
+ * or repositories directly from the controller.
+ *
+ * Route template inspection suppressions
+ *
+ * These inspections are failing because the {entity} segment of the RouteAttribute defined
+ * on the controller itself is not directly passed to each method. Instead of being passed to
+ * each method, it is bound to a property of the class, which is implicitly accessible to each method.
+ */
+
+[ApiController]
+[ApiVersion(1)]
+[ValidateAdminNoteControllerRoute]
+[Route("api/v{version:apiVersion}/{entity}")]
```

Comment:
Is there a reason why the entity is forced at this level? For endpoints that only rely on the primary key of the note directly, such as deletion, it shouldn't matter what the entity is.